KOKKOS_TOOLS_LIBS and kp_add_example for Kernel Filter

vlkale opened the pull request:
Kernel filter is used for nvtx_focused_connector and other tools. It is outdated given changes in user interface and cmake build systems. 
This PR is associated with #176 . 

This PR does the following: 

* Read the environment KOKKOS_TOOLS_LIBS in addition to KOKKOS_PROFILE_LIBRARY. 
* Use kp_add_example for Kernel Filter

vlkale committed:
kernel-filter: add_example to kp_add_example in its CMakeLists.txt

vlkale committed:
read KOKKOS_TOOLS_LIBS env variable

vlkale committed:
KernelFilter: apply clang format

dalg24 commented on common/kernel-filter/kp_kernel_filter.cpp:
```diff
@@ -126,7 +126,19 @@ extern "C" void kokkosp_init_library(const int loadSeq,
+      char* profileLibrary = getenv("KOKKOS_TOOLS_LIBS");
+      if (NULL == profileLibrary) {  // check for backward compatibility with
+                                     // old environment variable
+        printf(
+            "Checking KOKKOS_PROFILE_LIBRARY. WARNING: This is a deprecated "
+            "variable. Please use KOKKOS_TOOLS_LIBS.\n");
```
Why emit this warning if `KOKKOS_PROFILE_LIBRARY ` was not actually set?

vlkale replied:
Yeah, agreed - this is not a great warning text. It was copied and pasted from another library (nvtx-connector). The text after "WARNING: .." should be printed only in the case that KOKKOS_PROFILE_LIBRARY was set. I will change.

vlkale committed:
Emit env variable warning print only if not set 

Emit KOKKOS_PROFILE_LIBRARY environment variable warning print only if not set rather than printing whenever KOKKOS_TOOLS_LIBS is not found.

dalg24 commented on common/kernel-filter/kp_kernel_filter.cpp:
```diff
@@ -182,7 +196,7 @@ extern "C" void kokkosp_init_library(const int loadSeq,
     printf("============================================================\n");
   }
-}
+}  // end kokkosp_init_library
```
I don't really the point of marking the closing brace here.

dalg24 commented on common/kernel-filter/kp_kernel_filter.cpp:
```diff
@@ -126,7 +126,21 @@ extern "C" void kokkosp_init_library(const int loadSeq,
+         printf("KokkosP: Found that KOKKOS_PROFILE_LIBRARY is set and it will be used.\n");
+         printf("KokkosP: Note that KOKKOS_PROFILE_LIBRARY is a deprecated variable. " 
+         "Please use KOKKOS_TOOLS_LIBS in the future.\n");
+      }  // end check for backward compatability
```
Typo compat(i)bility.  I don't see the point of the comment. It is not even the right place.  You are missing a cling brace I don;'t even think this compiles

vlkale replied:
Thanks. I messed it up (in the process of trying to put that comment). See updated version .

Commit:
kernelFilter: fix parenthesis

Commit:
KernelFilter: applied clang format

vlkale commented:
Note that this PR doesn't use the namespace for KokkosTools and Filter and kp_kernel_filter.cpp has functions with extern "C" . This will be fixed in a separate PR.

Commit:
kernel_fiter: add cxx flags
